kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java

```diff
@@ -499,348 +532,174 @@ protected void handleInactive(KafkaHeaderAndRequest kafkaHeaderAndRequest,
-                        .whenComplete((isAuthorized, ex) -> {
-                            if (ex != null) {
-                                log.error("Describe namespace authorize failed, namespace - {}. {}",
-                                        namespace, ex.getMessage());
-                                completeOne.run();
-                                return;
-                            }
-                            if (isAuthorized) {
-                                pulsarService.getNamespaceService()
-                                        .getListOfPersistentTopics(NamespaceName.get(namespace))
-                                        .whenComplete((topics, e) -> {
-                                            if (e != null) {
-                                                log.error("Failed to getListOfPersistentTopic of {}", namespace, e);
-                                                topicMapFuture.completeExceptionally(e);
-                                                return;
-                                            }
-                                            if (topicMapFuture.isCompletedExceptionally()) {
-                                                return;
-                                            }
-                                            for (String topic : topics) {
-                                                final TopicName topicName = TopicName.get(topic);
-                                                final String key = topicName.getPartitionedTopicName();
-                                                topicMap.computeIfAbsent(
-                                                        KopTopic.removeDefaultNamespacePrefix(key, namespacePrefix),
-                                                        ignored -> Collections.synchronizedList(new ArrayList<>())
-                                                ).add(topicName);
-                                            }
-                                            completeOne.run();
-                                        });
-                            } else {
-                                completeOne.run();
-                            }
-                        });
             }
-        }).exceptionally(error -> {
-            topicMapFuture.completeExceptionally(error);
-            return null;
         });
-        return topicMapFuture;
+        return topicAndMetadataList;
+    }
+
+    private CompletableFuture<List<String>> authorizeNamespacesAsync(final Collection<String> namespaces,
+                                                                     final AclOperation aclOperation) {
+        final Map<String, CompletableFuture<Boolean>> futureMap = namespaces.stream().collect(
+                Collectors.toMap(
+                        namespace -> namespace,
+                        namespace -> authorize(aclOperation, Resource.of(ResourceType.NAMESPACE, namespace))
```

Comment:
handle exceptions when authorize namespaces.

Reply:
The reason is similar in the previous comment. There are two cases:

1. Some namespaces doesn't have the proper permission: printed the warning logs and ignore them

```java
            futureMap.entrySet().stream().filter(e -> {
                if (!e.getValue().join()) {
                    log.warn("Failed to authorize {} for ACL operation {}", e.getKey(), aclOperation);
                    return false;
                }
```
2. The authorization failed exceptionally. Go to the `getTopicsAsync`'s callback:

```java
        getTopicsAsync(request, fullTopicNameToOriginal.keySet()).whenComplete((topicAndMetadataList, e) -> {
            if (e != null) 
```